v3

davidmurdoch opened the pull request.

davidmurdoch commented on cli.js:
```diff
@@ -190,7 +190,10 @@ server.listen(options.port, options.hostname, function(err, result) {
   console.log("Listening on " + options.hostname + ":" + options.port);
-});
+} catch (error) {
+  console.log(error);
```
What do you think about updating this to `console.error`? Hmmmm. wonder if that'd break anyone's build if they check for errors on stdout instead of stderr. :-/

nicholasjpaterno replied:
That was my concern as well.

davidmurdoch commented on cli.js:
```diff
@@ -190,7 +190,10 @@ server.listen(options.port, options.hostname, function(err, result) {
+  return;
+}
 
 process.on('uncaughtException', function(e) {
```
Do you think these `process.on(...` handlers should go before the listen call now? Or doesn't it not matter since it's synchronous-ish now?

davidmurdoch commented on cli.js:
```diff
@@ -6,6 +6,7 @@ require('source-map-support').install();
 var yargs = require("yargs");
 var pkg = require("./package.json");
 var {toChecksumAddress, BN} = require("ethereumjs-util");
+var deasync = require('deasync');
```
<pedantic>Use double quotes here.

davidmurdoch commented:
I've updated this PR, but I'm still not confident in its stability. Might need to beta this one for a few weeks.

davidmurdoch commented on cli.js:
```diff
@@ -138,86 +143,100 @@ process.on("SIGINT", function () {
+  } else {
+    server.listen(options.port, options.hostname, startGanache);
+  }
+} catch (error) {
```
The `try`/`catch` here may cause a failure on Windows which is why I removed it in the previous commit.

davidmurdoch commented on README.md:
```diff
@@ -168,6 +168,7 @@ Both `.provider()` and `.server()` take a single object which allows you to spec
 * `"gasLimit"`: `String::hex | number` Sets the block gas limit. Must be specified as a `hex` string or `number`. Defaults to `"0x6691b7"`.
 * `"callGasLimit"`: `number` Sets the transaction gas limit for `eth_call` and `eth_estimateGas` calls. Must be specified as a `hex` string. Defaults to `"0x1fffffffffffff"` (`Number.MAX_SAFE_INTEGER`).
 * `"keepAliveTimeout"`:  `number` If using `.server()` - Sets the HTTP server's `keepAliveTimeout` in milliseconds. See the [NodeJS HTTP docs](https://nodejs.org/api/http.html#http_server_keepalivetimeout) for details. `5000` by default.
+* `"deasync"`:  `boolean` Synchronizes ganache server startup. Useful in certain scenarios - see ganache-cli issue [#733](https://github.com/trufflesuite/ganache-cli/issues/733).
```
Thoughts on changing to `--experimental-deasync` until we assess the viability?

nicholasjpaterno replied:
I wonder if it makes sense to just put experimental in the description so we dont break anything when we remove it?

davidmurdoch replied:
I was thinking we could always keep the `experimental` flag around as an alias.
Meh. Let's just ship it as is. :shipit:

tcoulter committed:
fix: add `debug_traceTransaction` RPC method (#614)

Co-authored-by: David Murdoch <david@davidmurdoch.com>

davidmurdoch committed:
fix: switch to `SecureTrie` (#665)

davidmurdoch committed:
fix: don't overwrite existing initial accounts (#666)

davidmurdoch committed:
test(eth_getCode): add tests for contract factories (#667)

davidmurdoch committed:
style: fix typo in a comment

davidmurdoch committed:
test(eth_sendTransaction): add contract deploy OOG test to ensure correct receipt values

davidmurdoch committed:
feat: add option to use gas estimate on send tx (#668)

davidmurdoch committed:
feat: add `legacyInstamine` flag to cli options

davidmurdoch committed:
feat: add `defaultTransactionGasLimit` flag to cli options

davidmurdoch committed:
3.0.0-internal.1

davidmurdoch committed:
fix: ensure tx errors return json-rpc code

davidmurdoch committed:
fix: include function and classname in minified build

davidmurdoch committed:
3.0.0-internal.3

davidmurdoch committed:
fix: change legacy option from networkId to network_id

davidmurdoch committed:
3.0.0-internal.4

davidmurdoch committed:
fix: continue network_id rename

davidmurdoch committed:
3.0.0-internal.5

davidmurdoch committed:
3.0.0-internal.6

davidmurdoch committed:
3.0.0-internal.7

davidmurdoch committed:
style: fix typo in comment

davidmurdoch committed:
refactor: make sendAsync & send call private fn

 This allows for monkey-patching send without affecting sendAsync, as Truffle currently does.

davidmurdoch committed:
v3.0.0-internal.8

davidmurdoch committed:
fix: fix bigint-buffer in build so it uses native lib

davidmurdoch committed:
v3.0.0-internal.9

davidmurdoch committed:
v3.0.0-internal.10

davidmurdoch committed:
fix: return rlp decoded values from getStorageAt

davidmurdoch committed:
v3.0.0.internal.11

davidmurdoch committed:
chore: add npm script to start cli

davidmurdoch committed:
test: correct miner test start up

davidmurdoch committed:
docs: update README.md path, assets, and docs

davidmurdoch committed:
docs: update docs for GH Pages (#679)

davidmurdoch committed:
docs: add polish

.

davidmurdoch committed:
refactor: rewrite into a TypeScript monorepo, partial (#657)

mikeseese committed:
chore: move chain packages to a subfolder (#703)

Co-authored-by: David Murdoch <david@davidmurdoch.com>

mikeseese committed:
feat: change CLI to dynamically generate args from flavor options (#708)

Co-authored-by: David Murdoch <david@davidmurdoch.com>

mikeseese committed:
chore: ignore folders that are packages from another branch (#725)

davidmurdoch committed:
chore: fix windows node 10.7.0 CI by updating npm (#731)

davidmurdoch committed:
chore: polish cli (#721)

Co-authored-by: Mike Seese <seesemichaelj@gmail.com>

davidmurdoch committed:
feat: handle mutually exclusive args in cli (#732)

mikeseese committed:
chore: lighten up github ci to enhance dev experience (#734)

Co-authored-by: David Murdoch <david@davidmurdoch.com>

mikeseese committed:
fix: update types on wallet legacy options (#737)

davidmurdoch committed:
chore: remove ubuntu-latest warning from action
